frontend/src/components/modals/ChannelModal.vue

```diff
@@ -25,7 +25,7 @@ const i18n = useI18n();
 const v = useVuelidate();
 
 const frozen = props.channel && props.channel.flags.includes(ChannelFlag.FROZEN);
-const possibleFlags = frozen ? [ChannelFlag.PINNED] : [ChannelFlag.UNSTABLE, ChannelFlag.PINNED];
+const possibleFlags = frozen ? [ChannelFlag.PINNED] : [ChannelFlag.UNSTABLE, ChannelFlag.PINNED, ChannelFlag.HIDE_BY_DEFAULT];
 
```

Comment:
I'm not really a fan of this hardcoded flags based on frozen. Probably should be part of the backend data, what channel flags are valid when frozen or not, but that's a separate PR.